src/vm/functions/mod.rs

```diff
@@ -68,10 +68,17 @@ define_named_enum!(NativeFunctions {
     ConsSome("some"),
     DefaultTo("default-to"),
     Asserts("asserts!"),
-    Expects("expects!"),
-    ExpectsErr("expects-err!"),
+    UnwrapRet("unwrap!"),
+    UnwrapErrRet("unwrap-err!"),
+    Unwrap("unwrap"),
+    UnwrapErr("unwrap-err"),
+    MatchOpt("match-opt"),
+    MatchResp("match-resp"),
```

Comment:
Hmm -- yeah, we could do that instead

Reply:
Sorry I'm late to the party.

If we go the `match` route over `match-opt` or `match-resp`, would this create a developer expectation that we can match on arbitrary values?  Is this a route we want to go down?  Not sure how complicated this would make the implementation.

Reply:
It wouldn't complicate the implementation very much -- it'd just add a rust match on the input type.

I also suspect it wouldn't mess with developer expectations very much --- ultimately it'll be type checked, so if someone tries to match on something else, it will fail. The only weirdness is that the match function signature is different in the two cases:

```
;; option case
;;  args are input, var-name for the some type, closure for some case, closure for none case
(match opt-input some-name (+ 1 some-name) 1)

;; response case
;; args are input, var-name for ok, closure for ok, var-name for err, closure for err
(match resp-input ok-name (ok (+ 1 ok-name)) err-name err-name)
```

I'm inclined towards going this route, since it makes the program a little simpler, and there's no real loss of specificity. I'll just make sure to add descriptive errors for the two cases.

Reply:
Ah, I was asking about whether or not `match` would create the expectation that you can match on any type, possibly with a variable number of match arms (e.g. matching on the value of a `uint`).

Reply:
Understood -- my point was that it will be pretty clear from documentation and the type checker that that wouldn't be supported.